### PR TITLE
New: Support for TLS 1.1 and 1.2 connections when .net 4.5 is installed

### PR DESCRIPTION
**Pull request by markus101**
#### Database Migration
NO

#### Description
Support for TLS 1.1 and TLS 1.2 when making HTTP requests. SSL3 and TLS are also still supported, SSL3 should be dropped for security reasons, but left a note to do that in v3.

### CODE REVIEW

File: src/NzbDrone.Host/Owin/OwinHostController.cs [hunk 1]
```diff
@@ -34,8 +34,6 @@ public class OwinHostController : IHostController
 
         public void StartServer()
         {
-            X509CertificateValidationPolicy.Register();
```
Comment (markus101) [inline src/NzbDrone.Host/Owin/OwinHostController.cs]: We're already doing this in Bootstrap.cs so it seems superfluous here.

File: src/NzbDrone.Common/Security/SecurityProtocolPolicy.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+namespace NzbDrone.Common.Security
+{
+    public static class SecurityProtocolPolicy
+    {
+        private static readonly Logger Logger = NzbDroneLogger.GetLogger(typeof(SecurityProtocolPolicy));
+
+        public static void Register()
+        {
+            Logger.Debug("Checking if TLS 1.1 and TLS 1.2 are supported");
+            if (Enum.IsDefined(typeof(SecurityProtocolType), 768) && Enum.IsDefined(typeof(SecurityProtocolType), 3072))
```
Comment (Taloth) [inline src/NzbDrone.Common/Security/SecurityProtocolPolicy.cs]: How about adding two `private const`, shall we?

File: src/NzbDrone.Host/Owin/OwinHostController.cs (hunk 1 above)
Comment (Taloth) [inline src/NzbDrone.Host/Owin/OwinHostController.cs]: Agreed

**Comment by Taloth** [comment]
Some constants would be nice. Now, how to test it? :wink:
Could also split TLS 1.1 and 1.2.

Try Catch might be good, afaik the property throws if you add invalid values, and it isn't critical.

The log entry isn't really useful, that only gets logged to console. (debug file isn't initialized yet)

**Pull request merged**